enzy_htp/_interface/xtb_interface.py

```diff
@@ -8,28 +8,241 @@
+
+    @property
+    def method(self) -> QMLevelOfTheory:
+        """Getter for the QMLevelOfTheory dictating the Engine's theoretical behavior."""
+        return self._method
+
+    @property
+    def geo_opt(self) -> bool:
+        """Is this a QMGeometryOptimizationEngine?"""
+        return self._geo_opt
+
+    @property
+    def parent_interface(self) -> XTBInterface:
+        """Getter for the Engine's parent interface."""
+        return self._parent_interface
+
+    @property
+    def work_dir(self) -> str:
+        """Getter for the work_dir where temporarily files are saved and commands are run."""
+        return self._work_dir
+
+    @property
+    def region(self) -> StructureRegion:
+        """Getter for the StructureRegion """
+        return self._region
+
+    @property
+    def engine(self) -> str:
+        """Name of the engine. Hardcoded to 'xtb'"""
+        return "xtb"
+
+    @property
+    def constraints(self) -> List[StructureConstraint]:
+        """Getter for the StructureConstraints owned by the Engine."""
+        return self._constraints
+
+    def make_job(self, stru: Structure) -> Tuple[ClusterJob, XTBQMResultEgg]:
+        """The method that makes a ClusterJob that runs the QM."""
+        if not isinstance(stru, Structure):
+            _LOGGER.error("The supplied variable stru MUST be a Structure()")
+            raise TypeError
+        
+        assert False, "Idk how to do this -CJ"
+
+    def run(self, stru: Structure) -> ElectronicStructure:
+        """Method that actually runs the XTB single point calculation. Returns results of calculation as an ElectronicStructure() object."""
+        if not isinstance(stru, Structure):
+            _LOGGER.error("Supplied stru is not of type Structure()")
+            raise TypeError()
+        
+        fs.safe_mkdir(self.work_dir)
+
+        sr = self.region
+        if sr is None:
+            sr = create_region_from_full_stru( stru )
+        
+        run_info:Dict = self.parent_interface.setup_xtb_run(
+            sr,
+            constraints=self.constraints,
+            lot=self.method,
+            geo_opt=self.geo_opt,
+            work_dir=self.work_dir,
+        )
+
+        start_dir:str = os.getcwd()
+        os.chdir(run_info['work_dir'])
+
+        results = self.parent_interface.env_manager_.run_command(
+                self.parent_interface.config()['XTB_EXE'],
+                run_info['args'])
+
+        os.chdir(start_dir)
+
+        lines:List[str] = fs.lines_from_file(run_info['xtb_outfile'])
+
+        return  ElectronicStructure(
+            energy_0 = self.parent_interface.parse_spe(lines),
+            geometry = stru,
+            mo = '',
+            mo_parser = '',
+            source="xtb",
+        )
+
+    def translate(self, result_egg: XTBQMResultEgg) -> ElectronicStructure:
+        """TODO(CJ)"""
+
+        assert False
+
+class XTBOptimizationEngine(XTBSinglePointEngine, QMOptimizationEngine):
```

Comment:
I make it this way because this is also how the RosettaQM way works. I also just noticed that there isn't much difference besides the fact that this type of run will also use constraints and update coordinates

Reply:
I think both could still inherit from `SimulationEngine`. I was just lazy and wrote a version o the `XTBInterface` functions that are the same for both run types and depend only on a flag for if they are geometry optimizations.

Reply:
Sounds good! I will change it when I make the `SimulationEngine`